src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -73,21 +143,45 @@ function useChatUtilsHook(): UseChatUtilsHook {
     });
   };
 
-  const goToChatInfoScreen = () => {
-    navigation.navigate('SampleChatInfoScreen');
+  const goToChatInfoScreen = (params?: object) => {
+    navigation.navigate('SampleChatInfoScreen', params);
   };
 
   const goBack = () => {
     navigation.goBack();
   };
+  const handleTextSystem = (item): string => {
+    if (
+      item?.rawJson?.userIdFollower === profile?.myProfile?.user_id ||
+      item?.rawJson?.message?.userIdFollower === profile?.myProfile?.user_id
+    ) {
+      return item?.rawJson?.textOwnMessage || item?.rawJson?.message?.textOwnMessage;
+    }
+    return item?.description || item?.message;
+  };
+
+  const splitSystemMessage = (message: string): string[] => {
+    const splitMessage = message?.split('.');
+    const pushMessage: string[] = [];
+    splitMessage?.forEach((systemMessage) => {
+      if (systemMessage && systemMessage.length > 0) {
+        const arrMessage: string = systemMessage?.trimStart()?.replace(/\n/g, '');
+        pushMessage.push(arrMessage);
+      }
+    });
+    return pushMessage;
+  };
 
   return {
     selectedChannel,
     goBack,
     goToChatScreen,
-    goToPostDetailScreen: helperGoToPostDetailScreen,
+    goToPostDetailScreen,
+    goToCommunityScreen,
     goToChatInfoScreen,
-    goBackFromChatScreen
+    goBackFromChatScreen,
+    handleTextSystem,
+    splitSystemMessage
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `handleTextSystem` function, there's a lot of optional chaining which can make the code hard to read and understand. Consider destructuring `item?.rawJson` and `profile?.myProfile` at the start of the function to simplify the code. Also, in the `splitSystemMessage` function, you're using `forEach` to iterate over `splitMessage` and push items into `pushMessage`. You could use `map` instead to make the code cleaner and more readable. Here's how you could refactor these functions:

```diff
  const handleTextSystem = (item): string => {
+      const { userIdFollower, textOwnMessage, message } = item?.rawJson || {};
+      const { user_id } = profile?.myProfile || {};

-     if (
-       item?.rawJson?.userIdFollower === profile?.myProfile?.user_id ||
-       item?.rawJson?.message?.userIdFollower === profile?.myProfile?.user_id
-     ) {
+     if (userIdFollower === user_id || message?.userIdFollower === user_id) {
-       return item?.rawJson?.textOwnMessage || item?.rawJson?.message?.textOwnMessage;
+       return textOwnMessage || message?.textOwnMessage;
      }
      return item?.description || item?.message;
    };

  const splitSystemMessage = (message: string): string[] => {
      const splitMessage = message?.split('.');
-     const pushMessage: string[] = [];
-     splitMessage?.forEach((systemMessage) => {
-       if (systemMessage && systemMessage.length > 0) {
-         const arrMessage: string = systemMessage?.trimStart()?.replace(/\n/g, '');
-         pushMessage.push(arrMessage);
-       }
-     });
-     return pushMessage;
+     return splitMessage?.map(systemMessage => 
+       systemMessage && systemMessage.trimStart().replace(/\n/g, '')
+     ).filter(Boolean) || [];
    };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->